internal/pkg/util/fs/helper.go

```diff
@@ -176,23 +176,33 @@ func MakeTmpFile(basedir, pattern string, mode os.FileMode) (*os.File, error) {
 	return f, nil
 }
 
-// CopyFile copies file to the provided location making sure the resulting
-// file has permission bits set to the mode.
-func CopyFile(from, to string, mode os.FileMode) (err error) {
-	dstFile, err := MakeTmpFile(filepath.Dir(to), "", mode)
+// FileExists simply checks if a path exists.
+func FileExists(path string) (bool, error) {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return false, nil
+	} else if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+// CopyFile copies file to the provided location. To honor umask
+// correctly, the to file must not exist.
+func CopyFile(from, to string, mode os.FileMode) error {
+	exist, err := FileExists(to)
```

Comment:
What if file is there with correct mode? 
I think this check should be performed in the calling function, rather than here.

Reply:
It is, but also double checked here. The code should never get here with `to` existing. But I can remove it if you want?